### PR TITLE
trigger reload configuration using SIGHUP

### LINKED ISSUE #537: Support SIGHUP signal for config reloading
With the [recent PR to add support for `/-/reload` endpoint](https://github.com/grafana/agent/issues/147), imo we should also allow for `SIGHUP` signals to follow the same codepath, such that we are consistent with [Prometheus's config reloading](https://github.com/prometheus/prometheus/blob/763a0d805c8f7f6e2cbd49c4c2c5e250e9dc5cd2/cmd/prometheus/main.go#L719).

**Expected**:
`kill -HUP $(sudo supervisorctl pid grafana-agent)` reloads the configuration

**Actual**
`kill -HUP $(sudo supervisorctl pid grafana-agent)` restarts the server completely

### PR DESCRIPTION
**Pull request by tharun208**
#### PR Description 
 This PR introduces reload of config using `SIGHUP`  signal. 

#### Which issue(s) this PR fixes 
Fixes #537.

#### PR Checklist

- [ ] CHANGELOG updated 
- [ ] Documentation added
- [ ] Tests updated

### CODE REVIEW

File: cmd/agent/entrypoint.go [hunk 1]
```diff
@@ -230,6 +233,9 @@ func (ep *Entrypoint) Start() error {
 	// signal is received.
 	signalHandler := signals.NewHandler(ep.cfg.Server.Log)
 
+	notifier := make(chan os.Signal)
+	signal.Notify(notifier, syscall.SIGHUP)
```
Comment (rfratto) [inline cmd/agent/entrypoint.go:237]: Can you defer a call to `signal.Stop` and `close(notifier)` here?

**Review by rfratto — COMMENTED** [review]
Thanks for working on this!

File: cmd/agent/entrypoint.go (hunk 1 above)
Comment (tharun208) [reply cmd/agent/entrypoint.go:237]: addressed

**Review by rfratto — APPROVED** [review]
Great, thank you!

**Comment by rfratto** [comment]
Oh, before I merge, could you add this to the CHANGELOG as an enhancement and give yourself credit? :)

**Comment by tharun208** [comment]
> Oh, before I merge, could you add this to the CHANGELOG as an enhancement and give yourself credit? :)

done 👍🏻

**Pull request merged**